### ISSUE TITLE
If an update request continues-as-new the SDK should complete the request with a specific response

### ISSUE THREAD

**Issue by Quinn-With-Two-Ns** [temporalio/sdk-java#1944]
**Is your feature request related to a problem? Please describe.**
If a users is using update to continue-as-new a workflow execution they probably want to know if the update request did successfully continue as new the workflow 

**Describe the solution you'd like**
Either fail or complete the workflow with a specific response

**Comment by bergundy** [comment]
I agree that ideally the failure would represent that the workflow continued as new, but IMHO, just dropping the update on the SDK and having the server take care of this makes sense. It would have to unblock any pending update, not just the one where the CAN call happened.

**Comment by Quinn-With-Two-Ns** [comment]
@bergundy I don't believe the server will fail/complete any outstanding updates when a workflow completes today. Should we move this issue to https://github.com/temporalio/temporal?

**Comment by Quinn-With-Two-Ns** [comment]
closing in favour of https://github.com/temporalio/temporal/issues/5174
